File: LeetCode/Python/0075. Sort Colors #2.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        c = collections.Counter(nums)
        
        for i in range(c[0]):
            nums[i] = 0
        for i in range(c[0], c[0] + c[1]):
            nums[i] = 1
        for i in range(c[0] + c[1], c[0] + c[1] + c[2]):
            nums[i] = 2

File: LeetCode/Java/0075. Sort Colors #2.java
class Solution {
    public void sortColors(int[] nums) {
        int zeroCount = 0;
        int oneCount = 0;
        int twoCount = 0;
        
        for (int num: nums) {
            if (num == 0) zeroCount++;
            else if (num == 1) oneCount++;
            else twoCount++;
        }
        
        for (int i = 0; i < zeroCount; i++)
            nums[i] = 0;
        for (int i = zeroCount; i < zeroCount + oneCount; i++)
            nums[i] = 1;
        for (int i = zeroCount + oneCount; i < zeroCount + oneCount + twoCount; i++)
            nums[i] = 2;
    }
}
